include/mc_control/MCController.h

```diff
@@ -262,8 +264,11 @@ struct MC_CONTROL_DLLAPI MCController
    * \return Vector of supported robots designed by name (as returned by
    * RobotModule::name())
    * \note
-   * Default implementation returns an empty list which indicates that all
+   * - Default implementation returns an empty list which indicates that all
    * robots are supported.
+   * - If the list is not empty, only the robots in that list are allowed to be
+   *   used with the controller. The main robot will be checked against the list of supported robots
+   * upon call to reset(), and an exception will be thrown if the robot is not supported.
    */
   virtual std::vector<std::string> supported_robots() const;
```

Comment:
Due to ABI boundaries issue, it's actually better if the method is:

```
virtual void supported_robots(std::vector<std::string> & out) const;
```

I think we can change it since that feature is not used much (or at all) outside of the samples

Reply:
Ok, done. 
Do you have a reference that explains why this is better?

Reply:
Passing a reference does not involve the ABI of `std::vector` but returning an `std::vector` cross the ABI boundary between mc_rtc and the shared library it loaded.

Note: generally this is a pretty bad idea to pass C++ types at ABI boundaries like this (even by reference), it kinda work in our case but that may broke for some combination of build flags between a controller library and mc_rtc itself. In the current version I think it kind-of works by accident (i.e. because we are lucky in the way we are manipulating the passed vector)